src/arrays.jl

```diff
@@ -21,35 +23,43 @@ function deepdiff(X::Vector, Y::Vector)
     # substrings.
 
     lengths = zeros(Int, length(X)+1, length(Y)+1)
+    backtracks = fill((0, 0), axes(lengths))
+    backtracks[1,2:end] .= Ref((0, 1))
+    backtracks[2:end,1] .= Ref((1, 0))
+    backtracks[1,1] = (0, 0)
 
     for (j, v2) in enumerate(Y)
         for (i, v1) in enumerate(X)
-            if v1 == v2
+            if isequal(v1, v2)
                 lengths[i+1, j+1] = lengths[i, j] + 1
+                backtracks[i+1, j+1] = (1, 1)
             else
-                lengths[i+1, j+1] = max(lengths[i+1, j], lengths[i, j+1])
+                lengths[i+1, j+1], backtracks[i+1, j+1] = _argmax(lengths[i+1, j], lengths[i, j+1])
```

Comment:
I don't think `_argmax` is a clear name. I'd just put that implementation here directly - this is the only place it's used, right?